File: python/basic_fizzbuzz.py
"""
A very basic and "typical" FizzBuzz with nothing fancy.

Explicitly checking for every possible match for every possible output.
"""

def basic_fizzbuzz(max_number: int = 30) -> None:
    for number in range(1, max_number + 1) :
        if number  % 15 == 0:
            print("FizzBuzz")
        elif number % 3 == 0:
            print("Fizz")
        elif number % 5 == 0:
            print("Buzz")
        else:
            print(number)

File: go/internal/fizzbuzz/basicFizzbuzz.go
package fizzbuzz

import "fmt"

func BasicFizzBuzz(maxNumber int) {
	for i := range maxNumber {
		i++
		if i%15 == 0 {
			fmt.Println("FizzBuzz")
		} else if i%3 == 0 {
			fmt.Println("Fizz")
		} else if i%5 == 0 {
			fmt.Println("Buzz")
		} else {
			fmt.Println(i)
		}
	}
}
